### PR TITLE
WedIDL: bring dom/bindings/typedarray further in line with spec

### LINKED ISSUE #31319: WedIDL: bring dom/bindings/typedarray further in line with spec
As we are refactoring our use of JS typed arrays(for example in https://github.com/servo/servo/pull/31202), I realized what we are doing--moving away from raw JSObject pointers to "typed stuff", and wrapping operations into structs that hide the use of `jsapi`-- we are actually simply following [the spec](https://webidl.spec.whatwg.org/); there are also things we can do to come even closer to it:

1. Rename [`typedarrays.rs`](https://github.com/servo/servo/blob/c367309a8fc004e47a6d8c0528b82e9d60a5716a/components/script/dom/bindings/typedarrays.rs) into `buffer_source_types.rs`(or similar), so that is matches [`idl-buffer-source-types`](https://webidl.spec.whatwg.org/#idl-buffer-source-types).
2. Wrap [the internal JSObject](https://github.com/servo/servo/blob/c367309a8fc004e47a6d8c0528b82e9d60a5716a/components/script/dom/bindings/typedarrays.rs#L18) into an enum whose variants would match that of [`BufferSource`](https://webidl.spec.whatwg.org/#BufferSource). 
3. Add links to the various operations, for example [`create_typed_array`](https://github.com/servo/servo/blob/c367309a8fc004e47a6d8c0528b82e9d60a5716a/components/script/dom/bindings/typedarrays.rs#L129) matches https://webidl.spec.whatwg.org/#arraybufferview-create (and we should rename the function to reflect that).
4. And so on. I think if you look at the spec it is quite clear what we can rename and link to to better match it.

### PR DESCRIPTION
**Pull request by Taym95**
As we are refactoring our use of JS typed arrays(for example in https://github.com/servo/servo/pull/31202), we realized what we are doing--moving away from raw JSObject pointers to "typed stuff", and wrapping operations into structs that hide the use of jsapi-- we are actually simply following [the spec](https://webidl.spec.whatwg.org/).

In this PR I renamed `typedarray` to `buffer_source` and added doc links to align with spec.

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #31319

- [x] These changes do not require tests because there is not behavior changes.

### CODE REVIEW

File: components/script/dom/bindings/buffer_source_types.rs [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+    BigInt64Array(Box<Heap<*mut JSObject>>),
+    BigUint64Array(Box<Heap<*mut JSObject>>),
+    Float32Array(Box<Heap<*mut JSObject>>),
+    Float64Array(Box<Heap<*mut JSObject>>),
+    DataView(Box<Heap<*mut JSObject>>),
+    ArrayBuffer(Box<Heap<*mut JSObject>>),
+    Default(Box<Heap<*mut JSObject>>),
+}
+
+pub struct HeapBufferSourceTypes<T> {
```
Comment (mrobinson) [inline components/script/dom/bindings/buffer_source_types.rs:41]: What does the `Types` here refer to? Maybe this should just be called `HeapBufferSource`?

**Review by mrobinson — COMMENTED** [review]
This looks good to me, but I have a question about what one name means.

File: components/script/dom/bindings/buffer_source_types.rs (hunk 1 above)
Comment (Taym95) [reply components/script/dom/bindings/buffer_source_types.rs:41]: I tried to follow spec blindly, indeed `Types` in not needed here

Comment (mrobinson) [reply components/script/dom/bindings/buffer_source_types.rs:41]: I would rename this to `HeapBufferSource` and this file to `buffersource` then to match the naming scheme we use for the bindings currently.

**Review by mrobinson — CHANGES_REQUESTED** [review]

**Review by mrobinson — APPROVED** [review]

**Pull request merged**